common/util/file_util.h

```diff
@@ -44,6 +44,10 @@ struct Directory {
 // empty string.
 absl::string_view Basename(absl::string_view filename);
 
+// Returns the part of the path after the final "/".  If there is no
+// "/" in the path, the result is the same as the input.
+absl::string_view Direname(absl::string_view filename);
```

Comment:
You mean "the path before", not "after"?
How about "return the directory that contains the file". And then mention the edge cases (what if it ends with "/", what about ".", "", "./"